src/store/slices/categoriesSlice.ts

```diff
@@ -0,0 +1,19 @@
+import { StateCreator } from 'zustand'
+
+export interface CategoriesSlice {
+  reducerState: {
+    activeCategory: number | null
+    activeSubCategory: number | null
+  }
+  setReducerState: (newState: { activeCategory: number | null, activeSubCategory: number | null }) => void
+}
+
+const createCategoriesSlice: StateCreator<CategoriesSlice> = (set) => ({
+  reducerState: {
+    activeCategory: null,
+    activeSubCategory: null,
+  },
+  setReducerState: (newState) => set({ reducerState: newState }),
```

Comment:
I don't like this approach. There's every single state cell in one object and one state dispatcher to update it.
Think about state this way:
```ts
    activeCategory: null,
    activeSubCategory: null,
    
    // method that describe an action instead of state that is being updated
    // eg. changeCategory: () => {
    // }
```